src/python/WMCore/ReqMgr/Service/Request.py

```diff
@@ -374,43 +374,55 @@ def _updateRequest(self, workload, request_args):
 
         dn = cherrypy.request.user.get("dn", "unknown")
 
-        if ('SoftTimeout' in request_args) and ('GracePeriod' in request_args):
-            request_args['HardTimeout'] = int(request_args['SoftTimeout']) + int(request_args['GracePeriod'])
-
-        if 'RequestPriority' in request_args:
-            self.gq_service.updatePriority(workload.name(), request_args['RequestPriority'])
+        # allow Priority update any time.
+        if 'RequestPriority' in request_args: 
+            self.gq_service.updatePriority(workload.name(), int(request_args['RequestPriority']))
 
         if "total_jobs" in request_args:
             # only GQ update this stats
             # request_args should contain only 4 keys 'total_jobs', 'input_lumis', 'input_events', 'input_num_files'}
             report = self.reqmgr_db_service.updateRequestStats(workload.name(), request_args)
-        # if is not just updating status
-        else:
-            req_status = request_args.get("RequestStatus", None)
-
-            if len(request_args) >= 1 and req_status == None:
-                try:
-                    workload.updateArguments(request_args)
-                except Exception as ex:
-                    msg = traceback.format_exc()
-                    cherrypy.log("Error for request args %s: %s" % (request_args, msg))
-                    raise InvalidSpecParameterValue(str(ex))
-                
-                # legacy update schema to support ops script
-                loadRequestSchema(workload, request_args)
-                # trailing / is needed for the savecouchUrl function
-                workload.saveCouch(self.config.couch_host, self.config.couch_reqmgr_db)
-
-            elif (req_status in ["closed-out"  "announced"]) and request_args.get("cascade", False):
-                cascade_list = self._retrieveResubmissionChildren(workload.name)
+            
+        if ('SoftTimeout' in request_args) and ('GracePeriod' in request_args):
+            request_args['HardTimeout'] = int(request_args['SoftTimeout']) + int(request_args['GracePeriod'])
+        
+        
+        req_status = request_args.get("RequestStatus", None)
+
+        if (len(request_args) >= 1 and req_status == None) or \
+            (len(request_args) > 1 and (req_status in ["assigned", "assignment-approved"])):
```

Comment:
Actually assignment-approved doesn't change anything now. but since something could change in assignment-approved possibly, I just added.

when there is req_status and len(request_args) == 1 we don't have to load workload and change its params. That is why I seperate the condition, for that case it will be handled below.